uint16[16] voltage
int16[16] diff
float32[16] initial_value
